src/Acuminator/Acuminator.Analyzers/StaticAnalysis/LegacyBqlField/LegacyBqlFieldFix.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Acuminator.Analyzers.StaticAnalysis.LegacyBqlField
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class LegacyBqlFieldFix : CodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1060_LegacyBqlField.Id);
+
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+			var classNode = root.FindNode(context.Span).FirstAncestorOrSelf<ClassDeclarationSyntax>();
+			if (classNode == null)
+				return;
+
+			var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1060_LegacyBqlField.Id);
+			if (diagnostic == null)
+				return;
+
+			if (diagnostic.Properties == null
+				|| !diagnostic.Properties.TryGetValue(LegacyBqlFieldAnalyzer.CorrespondingPropertyType, out string typeName)
+				|| String.IsNullOrEmpty(typeName))
+			{
+				return;
+			}
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			string title = nameof(Resources.PX1060Fix).GetLocalized().ToString();
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					title,
+					c => Task.FromResult(
+						context.Document.WithSyntaxRoot(
```

Comment:
@Derrekam  if you think that this code fix will be used in multiple place at once via Fix All in Solution/Project/Document then you should use Roslyn Simplifier to simplify names of the fields. It will be quite unconvenient to delete PX.Data.BQL namespace prefixes in the whole solution. I think ideally the fix should add the `using PX.Data.BQL` to the list of usings if there isn't one already.

Reply:
Is there an example in existing diagnostics that I can use?

Reply:
@Derrekam unfortunately there is only a little example in Acuminator in `SyntaxNodeGenerationUtils.GetAttributeList method. You can find more info here:
https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix (the end of the manual)
https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.simplification.simplifier?view=roslyn-dotnet

Reply:
Adding ```PX.Data.BQL``` namespace to a document may lead to some name conflicts in locations that are outside of current code fix scope. For example both ```PX.Data``` and ```PX.Data.BQL``` have ```Minus<TOperand>``` class

Reply:
Ok, then it will be to expensive to analyze for such conflicts.